### PR TITLE
Remove minimize and maximize default implementation

### LINKED ISSUE #1392: [modal-manager][slide-panel-manager] Remove the onMaximize and onMinimize defaulted implementation
# Bug Report

## Description

[Modal-Manager](https://engineering.cerner.com/terra-ui/components/terra-modal-manager/modal-manager/about) and [Slide-Panel-Manager](https://engineering.cerner.com/terra-ui/components/terra-slide-panel-manager/slide-panel-manager/slide-panel-manager) implement the `DisclosureManagerHeaderAdapter` header, which contains the 
`onMaximize` and `onMinimize` props and [implement](https://github.com/cerner/terra-framework/blob/main/packages/terra-modal-manager/src/ModalManager.jsx#L107-L108) them as assumed default behaviors, when they should not be assumed or defaulted. Modal content needing an expand to fullscreen behavior is not standard Cerner UX practice and needed only rarely. Same with slide-panels, which do not require an expand to fullscreen/full-width as a defaulted behavior. 

Having the expand/maximize icon always present in every modal and slide-panel in its current position without ability to opt in or out, for every modal that is presented creates a usability risk, since it is not expected by users and could easily create confusion or interruption.  Removing the implementation of the `onMaximize` and `onMinimize` props for the immediate term is recommended by UX.
 
Future enhancements will likely be addressing these problems, possibly by moving the location and changing the styling of the onMaximize/onMinimize buttons to better align with users' mental model of OS-style window frames and to be less disruptive, or more content-aware or implementation specific.

## Additional Context / Screenshots

Modal-Manager
Screen Shot 2021-03-30 at 5 14 49 PM

Slide-Panel-Manager
Screen Shot 2021-03-30 at 5 16 49 PM

## Expected Behavior

The expand to fullscreen is a rarely needed with current approaches, and can be implemented by content teams as a prop of the modal or slide-panel to initially disclose at fullscreen. Removing the implementation of the `onMaximize` and `onMinimize` props for the immediate term is recommended by UX.

## Possible Solution

Remove

https://github.com/cerner/terra-framework/blob/45ad1d99ed07912cd88170585c9bc431b6a034ca/packages/terra-modal-manager/src/ModalManager.jsx#L107-L108

and 

https://github.com/cerner/terra-framework/blob/45ad1d99ed07912cd88170585c9bc431b6a034ca/packages/terra-slide-panel-manager/src/SlidePanelManager.jsx#L104-L105

### PR DESCRIPTION
**Pull request by BenBoersma**
### Summary
Removed the default implementation of onMinimize and onMaximize. Regenerated screenshots and removed some tests specifically testing for the Maximize and Minimize buttons.

Closes #1392 

### Deployment Link

https://terra-framew-issue-1392-zviioo.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by benbcai** [comment]
Should the [modal-manager documentation](https://engineering.cerner.com/terra-ui/components/terra-modal-manager/modal-manager/about#disclosuremanagerheaderadapter-support) and [slide-panel-manager documentation](https://engineering.cerner.com/terra-ui/components/terra-slide-panel-manager/slide-panel-manager/slide-panel-manager#disclosuremanagerheaderadapter-support) be updated to not reference maximize/minimize functionalities?

**Comment by BenBoersma** [comment]
> Should the [modal-manager documentation](https://engineering.cerner.com/terra-ui/components/terra-modal-manager/modal-manager/about#disclosuremanagerheaderadapter-support) and [slide-panel-manager documentation](https://engineering.cerner.com/terra-ui/components/terra-slide-panel-manager/slide-panel-manager/slide-panel-manager#disclosuremanagerheaderadapter-support) be updated to not reference maximize/minimize functionalities?

Removed here https://github.com/cerner/terra-framework/pull/1408/commits/56bdc4e2bdd46b4e97726b380bafd48ef06a21b5

File: packages/terra-modal-manager/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Removed
+  * Removed the onMaximize and onMinimize defaulted implementation
```
Comment (yuderekyu) [inline packages/terra-modal-manager/CHANGELOG.md:6]: nitpick - can we say *why* we are removing these?

File: packages/terra-slide-panel-manager/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Removed
+  * Removed the onMaximize and onMinimize defaulted implementation
```
Comment (yuderekyu) [inline packages/terra-slide-panel-manager/CHANGELOG.md:6]: same comment here

File: packages/terra-modal-manager/CHANGELOG.md (hunk 1 above)
Comment (BenBoersma) [reply packages/terra-modal-manager/CHANGELOG.md]: updated here https://github.com/cerner/terra-framework/pull/1408/commits/7d7c2fda45f4cfa167f07897be9178a96fd3abc4

**Review by yuderekyu — COMMENTED** [review]
are we defining this as a bug fix? i could see someone arguing that this is a non passive change

**Comment by mjhenkes** [comment]
@yuderekyu, yeah we're defining this as a defect. Some how we got wires crossed when implementing this but it was not intended behaviour.

**Review by mjhenkes — APPROVED** [review]

**Review by yuderekyu — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Review by neilpfeiffer — APPROVED** [review]
Tested: PR #1408
Resolves: Issue #1392

**Reviewed:**
- Correctly removes the onMaximize/onMinimize buttons as the default implementation to all modals and slide panels, correcting the usability risks that it creates for users.
- Screenshot updates all appear correct in these and dependent components


 **UX Reviewed** |
-----------|

----
\
\
Future fixes / possible adjustments to re-address this change, notes from originating issue #1392:

> Future enhancements will likely be addressing these problems, possibly by moving the location and changing the styling of the onMaximize/onMinimize buttons to better align with users' mental model of OS-style window frames and to be less disruptive, or more content-aware or implementation specific.

**Review by tbiethman — APPROVED** [review]

**Comment by chrismichalewicz** [comment]
Functionally reviewed.

**Pull request merged**